javascript/ql/experimental/adaptivethreatmodeling/lib/experimental/adaptivethreatmodeling/EndpointCharacteristics.qll

```diff
@@ -7,10 +7,46 @@ private import semmle.javascript.security.dataflow.SqlInjectionCustomizations
 private import semmle.javascript.security.dataflow.DomBasedXssCustomizations
```

Comment:
This file has (unsurprisingly) gotten quite big, since it now holds all the characteristics we use to make decisions about endpoint selection for training and inference. I'm happy to break it up into a few modules, and would love to hear suggestions on a good breakdown to keep things clear and well-organized. I currently have it broken into sections (separated by `/*...*/` comments) based on what the different characteristics have historically been used for (endpoint filtering, selecting negative training samples, etc.), but I'm hesitant to break it into modules based on these historical considerations, because the whole point of this revamp was to make it easy to experiment with changing these considerations. One idea would be to leave it as one file for now, do some experimentation to see what changes we want to make to the characteristics/implications, and then, once the dust settles, decide what kind of breakdown makes sense 🤷

Reply:
It may become necessary to break query-specific characteristics out into their own query specific files. But we'll cross that bridge when we get there, I'd say.